### PR TITLE
sc-52315/update-eslint-leadconduit-types

### PR DESCRIPTION
**Pull request by johnrb2**
## Description of the change

> bump eslint to new versions, bump mocha, run eslint --fix .

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change
- [x] Technical Debt
- [ ] Documentation

## Related tickets

> https://app.shortcut.com/active-prospect/story/52315/update-eslint-leadconduit-types

### Development and Testing

- [x]  Lint rules pass locally.
- [x]  The code changed/added as part of this pull request has been covered with tests, or this PR does not alter production code.
- [x]  All tests related to the changed code pass in development, or tests are not applicable.

### Code Review

- [x]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached.
- [x]  At least two engineers have been added as "Reviewers" on the pull request.
- [x]  Changes have been reviewed by at least two other engineers who did not write the code.
- [x]  This branch has been rebased off master to be current.

### Tracking 
- [x]  Issue from Shortcut/Jira has a link to this pull request.
- [x]  This PR has a link to the issue in Shortcut.

### QA
- [x]  This branch has been deployed to staging and tested.

### CODE REVIEW

File: test/normalize.test.js [hunk 1]
```diff
@@ -102,7 +102,7 @@ describe('Normalize utility', function () {
   });
 
   it('should normalize array', function () {
-    const arr = [new String('a string'), new Number(8), new Boolean(true)];
+    const arr = [new String('a string'), new Number(8), new Boolean(true), undefined];
```
Comment (johnrb2) [inline test/normalize.test.js:105]: added undefined to array to get code coverage back up

File: lib/normalize.js [hunk 2]
```diff
@@ -25,6 +25,7 @@ const normalize = function (obj) {
   } else if (obj instanceof Array) {
     obj = _.compact(obj.map((i) => {
       if (typeof i !== 'function') { return normalize(i); }
+      return ''; // will be removed by _.compact()
```
Comment (johnrb2) [inline lib/normalize.js:28]: updated test below for this line change

**Review by jack-ap — DISMISSED** [review]

File: test/normalize.test.js [hunk 3]
```diff
@@ -102,7 +102,8 @@ describe('Normalize utility', function () {
   });
 
   it('should normalize array', function () {
-    const arr = [new String('a string'), new Number(8), new Boolean(true)];
+    const emptyFunction = () => {};
+    const arr = [new String('a string'), new Number(8), new Boolean(true), emptyFunction];
```
Comment (johnrb2) [inline test/normalize.test.js:106]: added to bring back up code coverage

**Review by richardfordjunior — APPROVED** [review]
LGTM!

**Review by jack-ap — APPROVED** [review]

**Pull request merged**